### PR TITLE
Fix error where import * tried to overwrite a macro

### PR DESCRIPTION
**Pull request by da-woods**
Closes https://github.com/cython/cython/issues/4927

### CODE REVIEW

**Comment by da-woods** [comment]
Closed/opened to rerun against 0.29.x.

I'm not convinced this'll work cleanly on master (`noexcept` isn't yet in...) and it may well cause some test suite warning line number changes. I'm away from my PC for a bit so if you want it merged quickly then feel free to push the fixes

File: tests/memoryview/memslice.pyx [hunk 1]
```diff
@@ -21,6 +21,12 @@ if sys.version_info[0] < 3:
 else:
     import builtins
 
+try:
+    from this_module_doesnt_exist import *
```
Comment (scoder) [inline tests/memoryview/memslice.pyx:25]: Let's use something that we can at least control.
```suggestion
    from Cython.Tests.this_module_does_not_exist import *
```

File: Cython/Utility/MemoryView_C.c [hunk 2]
```diff
@@ -24,6 +24,9 @@ typedef struct {
 #ifndef CYTHON_ATOMICS
     #define CYTHON_ATOMICS 1
 #endif
+// using CYTHON_ATOMICS as a cdef extern bint in the Cython memoryview code
+// interacts badly with "import *". Therefore, define a helper function-like macro
+#define __PYX_CYTHON_ATOMICS_ENABLED() CYTHON_ATOMICS
```
Comment (scoder) [inline Cython/Utility/MemoryView_C.c:29]: Can we move this #define into a verbatim code section in `MemoryView.pyx`? (Assuming that works…) It really only exists to make up for a some problematic behaviour over there – which might even go away at some point.

**Review by scoder — COMMENTED** [review]
Thanks for figuring this out and working on a fix so quickly.

File: tests/memoryview/memslice.pyx (hunk 1 above)
Comment (da-woods) [reply tests/memoryview/memslice.pyx]: For what it's worth it doesn't matter if the import fails or not so we could test a successful import instead. I just thought that a failed one was less likely to introduce unwanted names.

File: Cython/Utility/MemoryView_C.c (hunk 2 above)
Comment (scoder) [reply Cython/Utility/MemoryView_C.c:29]: Lovely. Let's keep it as it is then.

**Comment by scoder** [comment]
Thanks

**Pull request merged**